AxisMovementConstraint.cs

```diff
@@ -0,0 +1,156 @@
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.SpatialManipulation;
+using UnityEngine;
+
+public class AxisMovementConstraint : TransformConstraint
+{
+    #region Properties
+
+    [SerializeField] 
+    [Tooltip("Allow movement along an X-Axis")]
+    private bool movementOnXAxis;
+
+
+    [SerializeField]
+    [Tooltip("Allow movement along an Y-Axis")]
+    private bool movementOnYAxis;
+
+
+    [SerializeField]
+    [Tooltip("Allow movement along an Z-Axis")]
+    private bool movementOnZAxis;
+
+    [SerializeField]
+    [Tooltip("Set the size of the steps for fixed movement")]
+    private float fixedStepSize = 1.0f;
+
+    [SerializeField]
+    [Tooltip("Enable to allow re-positioning only in fixed steps.")]
+    private bool bFixedStep = false;

```

Comment:
style: avoid using Hungarian notation in variable name (style guide pending) Remove b in bFixedStep